src/main/scala/org/apache/spark/sql/rikai/expressions/Box2d.scala

```diff
@@ -45,7 +45,11 @@ case class Area(child: Expression)
 
   override def prettyName: String = "area"
 
-  @enableIf(scala.util.Properties.versionNumberString.compareTo("2.12.15") >= 0)
+  @enableIf(c =>
+    c.classPath.exists(
+      _.getPath.matches(".*spark-catalyst_2\\.\\d+-3\\.2\\..*")
```

Comment:
These rules seems to be the same? shall we just extract them into one place, like `object SparkVersion` or something?

Reply:
It's a compile-time plugin, I'm not sure if it can access the source code when it's compiling. A practical way is to put this code into a submodule and to make the submodule a compiler plugin of other modules, which will increase the complexity of build-time work, and may also let IDEs in trouble.